src/ray/gcs/redis_module/ray_redis_module.cc

```diff
@@ -39,6 +39,9 @@ extern RedisChainModule module;
     }                                                            \
   }
 
+/// Map from pub sub channel to clients that are waiting on that channel.
+std::unordered_map<std::string, std::vector<std::string>> notification_map;
```

Comment:
Can we add the size of this to the debug string somehow?

Reply:
We could alternatively print a warning if this map exceeds certain size bounds, e.g. 1000, 10k, 100k.

Reply:
done, I added a redis command that can query the DEBUG_STRING. Also found a bug doing so, which was there before this PR: https://github.com/ray-project/ray/pull/4021/commits/4c62c3b01faa7ca6372eac87e1485902ca43b525